From 58446ec8b14b7b3a749975080629c3f3dadf5092 Mon Sep 17 00:00:00 2001
From: Chris Richards <chris.richards@yellowfeather.co.uk>
Date: Wed, 21 Mar 2012 10:34:55 -0500
Subject: [PATCH] ARM omap am335x BeagleBone userspace SPI support

SPI working on the Beagleboard XM rev C
http://www.brianhensley.net/2012/02/spi-working-on-beagleboard-xm-rev-c.html

BeagleBone userspace SPI interface (spidev) patch
https://groups.google.com/forum/#!topic/beagleboard/B3akyoyjwG4/discussion
I was unable to apply the patch at the beginning of this discussion

Signed-off-by: Chris Richards <chris.richards@yellowfeather.co.uk>
Acked-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/mach-omap2/board-am335xevm.c |   19 +++++++++++++++++++
 1 files changed, 19 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 87a52fc..a81e7f0 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -1592,6 +1592,16 @@ static struct gpmc_timings am335x_nand_timings = {
 	.wr_data_mux_bus = 0,
 };
 
+static struct spi_board_info bone_am335x_slave_info[] = {
+	{
+		.modalias      = "spidev",
+		.irq           = -1,
+		.max_speed_hz  = 12000000,
+		.bus_num       = 2,
+		.chip_select   = 0,
+	},
+};
+
 static void evm_nand_init(int evm_id, int profile)
 {
 	struct omap_nand_platform_data *pdata;
@@ -2183,6 +2193,14 @@ static void spi1_init(int evm_id, int profile)
 	return;
 }
 
+/* setup beaglebone spi1 */
+static void bone_spi1_init(int evm_id, int profile)
+{
+	setup_pin_mux(spi1_pin_mux);
+	spi_register_board_info(bone_am335x_slave_info,
+			ARRAY_SIZE(bone_am335x_slave_info));
+	return;
+}
 
 static int beaglebone_phy_fixup(struct phy_device *phydev)
 {
@@ -2351,6 +2369,7 @@ static struct evm_dev_cfg beaglebone_dev_cfg[] = {
 	{i2c2_init,	DEV_ON_BASEBOARD, PROFILE_NONE},
 	{boneleds_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
 	{bonew1_gpio_init, DEV_ON_BASEBOARD, PROFILE_ALL},
+	{bone_spi1_init,	DEV_ON_BASEBOARD, PROFILE_ALL},
 	{NULL, 0, 0},
 };
 
-- 
1.7.9.1

